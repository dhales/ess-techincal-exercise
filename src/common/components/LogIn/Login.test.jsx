import React from 'react';

import {render, getByTestId} from '@testing-library/react';

import { UserContext } from '../../state/context';

import Login from './LogIn';

const user = {
    userName:null, auth:false
}

const updateUser = () => null;

describe("Login modal", () => {
    it('Should render modal', () => {
        const { getByTestId } = render(
            <UserContext.Provider value={user, updateUser}>
                <Login 
                    closeCallBack={()=>null}
                    isOpen={true}
                />
            </UserContext.Provider>
            
        );
        const conatiner = getByTestId("login-modal");
        expect(conatiner).toMatchSnapshot();
    });
});