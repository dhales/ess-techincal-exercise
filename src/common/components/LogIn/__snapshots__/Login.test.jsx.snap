// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login modal Should render modal 1`] = `
<div
  aria-describedby="transition-modal-description"
  aria-labelledby="transition-modal-title"
  class="makeStyles-modal-2"
  data-testid="login-modal"
  role="presentation"
  style="position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px;"
>
  <div
    aria-hidden="true"
    class="MuiBackdrop-root"
    style="opacity: 1; webkit-transition: opacity 500ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 500ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
  />
  <div
    data-test="sentinelStart"
    tabindex="0"
  />
  <div
    class="makeStyles-paper-3"
    style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    tabindex="-1"
  >
    <div
      class="MuiTypography-root MuiTypography-h5"
    >
      Login Form
    </div>
    <form
      autocomplete="off"
      class="makeStyles-root-1"
      novalidate=""
    >
      <div
        class="MuiFormControl-root MuiTextField-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled"
          data-shrink="false"
          for="filled-basic"
          id="filled-basic-label"
        >
          Enter your user name
        </label>
        <div
          class="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiFilledInput-input"
            id="filled-basic"
            type="text"
            value=""
          />
        </div>
        
      </div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained Mui-disabled Mui-disabled"
        disabled=""
        tabindex="-1"
        type="submit"
      >
        <span
          class="MuiButton-label"
        >
          Submit
        </span>
      </button>
    </form>
  </div>
  <div
    data-test="sentinelEnd"
    tabindex="0"
  />
</div>
`;
